### ISSUE TITLE
AWS ASG Scheduling Support

### ISSUE THREAD

**Issue by dkhunt27** [hashicorp/terraform#2367]
Any plans to add support in the AWS provider for add/update a schedule for an AutoScalingGroup?  Seems to be the only functionality that we consistently use that terraform doesn't support.

http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/AutoScaling.html#putScheduledUpdateGroupAction-property

**Comment by ajmath** [comment]
This would be great for my team.  We have a cluster that grows in size over night to do batch processing and would like to utilize this.  There is an ongoing discussion on the google group including a workaround that uses the awscli in a local provisioner.

https://groups.google.com/forum/#!topic/terraform-tool/EBWHxTtbbtk

**Comment by ajmath** [comment]
This is something like what I'm envisioning:

```HCL
resource "aws_autoscaling_group" "bar" {
  availability_zones = ["us-east-1a"]
  max_size = 15
  min_size = 2
  launch_configuration = "${aws_launch_configuration.bar.name}"
}

/* Scale up and down on a daily schedule */
resource "aws_autoscaling_scheduled_scaling" "reoccurring_scale_up" {
  autoscaling_group_name = "${aws_autoscaling_group.bar.name}"
  recurrence = "0 19 * * *" 
  desired_capacity = 8
}
resource "aws_autoscaling_scheduled_scaling" "reoccurring_scale_down" {
  autoscaling_group_name = "${aws_autoscaling_group.bar.name}"
  recurrence = "0 5 * * *" 
  desired_capacity = 2
}

/* Schedule exact times for scaling */
resource "aws_autoscaling_scheduled_scaling" "cyber_monday_boost_start" {
  autoscaling_group_name = "${aws_autoscaling_group.bar.name}"
  start_time = "2015-11-29T22:00:00Z" 
  desired_capacity = 15
}
resource "aws_autoscaling_scheduled_scaling" "cyber_monday_boost_end" {
  autoscaling_group_name = "${aws_autoscaling_group.bar.name}"
  start_time = "2015-12-01T01:00:00Z" 
  desired_capacity = 8
}
```
The AWS documentation for this is at http://docs.aws.amazon.com/AutoScaling/latest/DeveloperGuide/schedule_time.html

**Comment by dkhunt27** [comment]
I am using something very similar....I just created a wrapper around the node aws sdk...but my "inputs" look like

{
        "method": "putScheduledUpdateGroupAction",
        "params": {
          "AutoScalingGroupName": "{{autoScaleGroupName}}",
          "ScheduledActionName": "{{ScheduledActionNamePrefix}}-goToSleep",
          "MinSize": 0,
          "DesiredCapacity": 0,
          "MaxSize": 5,
          "Recurrence": "0 0 * * *"
        }
      },
      {
        "method": "putScheduledUpdateGroupAction",
        "params": {
          "AutoScalingGroupName": "{{autoScaleGroupName}}",
          "ScheduledActionName": "{{ScheduledActionNamePrefix}}-wakeUp",
          "MinSize": 0,
          "DesiredCapacity": 1,
          "MaxSize": 5,
          "Recurrence": "0 12 * * 1-5"
        }
      }

**Comment by mcansky** [comment]
:+1:

**Comment by stack72** [comment]
FYI, I am in the process of adding this #4256

**Comment by jen20** [comment]
This has landed in #4256! Thanks @stack72!
